python/app/action_manager.py

```diff
@@ -0,0 +1,196 @@
+                                                                sg_data=sg_data, 
+                                                                actions=actions_to_evaluate,
+                                                                ui_area=ui_area_str)
+                except Exception:
+                    self._app.log_exception("Could not execute generate_actions hook.")
+            
+        # create QActions
+        actions = []
+        for action_def in action_defs:
+            name = action_def["name"]
+            caption = action_def["caption"]
+            params = action_def["params"]
+            description = action_def["description"]
+            
+            a = QtGui.QAction(caption, None)
+            a.setToolTip(description)
+            a.triggered[()].connect(lambda n=name, sg=sg_data, p=params: self._execute_hook(n, sg, p))
+            actions.append(a)
+            
+        if ui_area == self.UI_AREA_DETAILS:
+            actions = self._get_default_detail_actions(sg_data) + actions
+            
+        return actions
+
+    def _get_default_detail_actions(self, sg_data):
+        """
+        Returns a list of default actions for the detail area
+        """
+        refresh = QtGui.QAction("Refresh", None)
+        refresh.triggered[()].connect(lambda f=sg_data: self._refresh(f))
+
+        view_in_sg = QtGui.QAction("View in Shotgun", None)
+        view_in_sg.triggered[()].connect(lambda f=sg_data: self._show_in_sg(f))
+
+        copy_url = QtGui.QAction("Copy Shotgun url to clipboard", None)
+        copy_url.triggered[()].connect(lambda f=sg_data: self._copy_to_clipboard(f))
+
+        separator = QtGui.QAction(None)
+        separator.setSeparator(True)
+        
+        return [refresh, view_in_sg, copy_url, separator]
+    
+    ########################################################################################
+    # callbacks
+    
+    def _execute_hook(self, action_name, sg_data, params):
+        """
+        callback - executes a hook
+        """
+        self._app.log_debug("Calling scene load hook for %s. Params: %s. Sg data: %s" % (action_name, params, sg_data))
+        
+        try:
+            self._app.execute_hook_method("actions_hook", 
+                                          "execute_action", 
+                                          name=action_name, 
+                                          params=params, 
+                                          sg_data=sg_data)
+        except Exception, e:
+            self._app.log_exception("Could not execute execute_action hook.")
+            QtGui.QMessageBox.critical(None, "Hook Error", "Error: %s" % e)
+
+    def _refresh(self, entity):
+        """
+        Internal action callback - refreshes the main dialog UI
+        
+        :param entity: std sg entity dict with keys type, id and name
+        """
+        self._dialog.setup_ui()
```

Comment:
yes, that's a much better approach! Hmm, i think this code was part of those siggraph sessions....